File: demo/DemoMongoDB/DemoMongoDB_FE/src/services/productService.js
import axios from 'axios'

const API_BASE_URL = 'http://localhost:8080'

const productService = {
  getAllProducts: () => {
    return axios.get(`${API_BASE_URL}/products`)
  },

  getProductById: (id) => {
    return axios.get(`${API_BASE_URL}/products/${id}`)
  },

  createProduct: (productData) => {
    return axios.post(`${API_BASE_URL}/products`, productData)
  },

  updateProduct: (id, productData) => {
    return axios.put(`${API_BASE_URL}/products/${id}`, productData)
  },

  deleteProduct: (id) => {
    return axios.delete(`${API_BASE_URL}/products/${id}`)
  },

  getProductsByBrand: (brand) => {
    return axios.get(`${API_BASE_URL}/products/brand/${brand}`)
  },

  getProductsByCategory: (category) => {
    return axios.get(`${API_BASE_URL}/products/category/${category}`)
  },

  getProductsByPriceRange: (min, max) => {
    return axios.get(`${API_BASE_URL}/products/price`, {
      params: { min, max }
    })
  }
}

export default productService

File: demo/DemoMongoDB/DemoMongoDB_BE/src/main/java/fu/se/sba301/phongtt/demomongodb_be/service/ProductService.java
package fu.se.sba301.phongtt.demomongodb_be.service;



import fu.se.sba301.phongtt.demomongodb_be.document.Product;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
@RequiredArgsConstructor
public class ProductService {

    private final fu.se.sba301.phongtt.demomongodb_be.repository.ProductRepository productRepository;

    // GET ALL
    public List<Product> getAllProducts() {
        return productRepository.findAll();
    }

    // GET BY ID
    public Optional<Product> getProductById(String id) {
        return productRepository.findById(id);
    }

    // CREATE
    public Product createProduct(Product product) {
        return productRepository.save(product);
    }

    // UPDATE
    public Product updateProduct(String id, Product product) {
        product.setId(id);
        return productRepository.save(product);
    }

    // DELETE
    public void deleteProduct(String id) {
        productRepository.deleteById(id);
    }

    // FIND BY BRAND
    public List<Product> getByBrand(String brand) {
        return productRepository.findByBrand(brand);
    }

    // FIND BY CATEGORY
    public List<Product> getByCategory(String category) {
        return productRepository.findByCategory(category);
    }

    // FIND BY PRICE RANGE
    public List<Product> getByPriceRange(Double min, Double max) {
        return productRepository.findByPriceBetween(min, max);
    }
}